src/main/java/seedu/address/storage/JsonAdaptedInternshipApplication.java

```diff
@@ -133,11 +145,27 @@ public InternshipApplication toModelType() throws IllegalValueException {
             final Email modelEmail = new Email(contact.get(1));
             final Contact modelContact = new Contact(modelPhone, modelEmail);
 
+            if (documents.size() == 2) {
+                if (!ResumeLink.isValidResumeLink(documents.get(0))) {
+                    throw new IllegalValueException(ResumeLink.MESSAGE_CONSTRAINTS);
+                }
+                final ResumeLink modelResumeLink = new ResumeLink(documents.get(0));
+
+                if (!CoverLetterLink.isValidCoverLetterLink(documents.get(1))) {
+                    throw new IllegalValueException(CoverLetterLink.MESSAGE_CONSTRAINTS);
+                }
+                final CoverLetterLink modelCoverLetterLink = new CoverLetterLink(documents.get(1));
+                final Documents modelDocuments = new Documents(modelResumeLink, modelCoverLetterLink);
+
+                return new InternshipApplication(modelCompanyName, modelJobTitle, modelReviews, modelContact,
+                        modelStatus, modelInterviewDate, modelDocuments);
+            }
+
```

Comment:
This method may be too long. Since this method has no parameter, I believe that it will not be difficult to extract out each portion as a private helper method. Could you help us to do this?

Reply:
Yes definitely!